integration-tests/shard-operator-integration-tests/shard-operator-integration-tests-v2/src/test/java/com/redhat/service/smartevents/shard/operator/v2/cucumber/steps/ProcessorResourceSteps.java

```diff
@@ -0,0 +1,94 @@
+import io.cucumber.java.en.When;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+
+public class ProcessorResourceSteps {
+
+    private TestContext context;
+
+    public ProcessorResourceSteps(TestContext context) {
+        this.context = context;
+    }
+
+    @When("^deploy ProcessorResource:$")
+    public void deployProcessorResource(String processorResourceYaml) throws InterruptedException {
+        String filteredProcessorResourceYaml = ContextResolver.resolveWithScenarioContext(context, processorResourceYaml);
+        InputStream resourceStream = new ByteArrayInputStream(filteredProcessorResourceYaml.getBytes(StandardCharsets.UTF_8));
+        ManagedProcessor managedProcessor = context.getClient()
+                .resources(ManagedProcessor.class)
+                .inNamespace(context.getNamespace())
+                .load(resourceStream)
+                .createOrReplace();
+    }
+
+    @Then("^the ProcessorResource \"([^\"]*)\" exists within (\\d+) (?:minute|minutes)$")
+    public void theProcessorResourceExistsWithinMinutes(String name, int timeoutMinutes) throws TimeoutException {
+        Awaitility.await()
+                .atMost(Duration.ofMinutes(timeoutMinutes))
+                .pollInterval(Duration.ofSeconds(5))
+                .until(() -> {
+                    ManagedProcessor processor = context.getClient().resources(ManagedProcessor.class).inNamespace(context.getNamespace()).withName(name).get();
+                    return processor != null;
+                });
+    }
+
+    @Then("^the Deployment \"([^\"]*)\" is ready within (\\d+) (?:minute|minutes)$")
+    public void theDeploymentIsInStateWithinMinutes(String name, int timeoutInMinutes) throws TimeoutException {
+        Awaitility.await()
+                .atMost(Duration.ofMinutes(timeoutInMinutes))
+                .pollInterval(Duration.ofSeconds(5))
+                .until(
+                        () -> {
+                            Deployment deployment = context.getClient().apps().deployments().inNamespace(context.getNamespace()).withName(name).get();
+                            if (deployment == null) {
+                                return false;
+                            }
+                            //I am not sure if this code is valid for v2: deployment.getStatus().getConditions().stream().anyMatch(d -> d.getType().equals("Available") && d.getStatus().equals("True"))
+                            //Th PR of MGDOBR-1248 contains this implementation of the `isReadyV2` function in CustomResourceStatus: conditions.stream().allMatch(c -> ConditionStatus.True.equals(c.getStatus()))
+                            //return deployment.getStatus().getConditions().stream().allMatch(d -> d.getStatus().equals("True"));
+                            return deployment.getStatus().getConditions().stream().anyMatch(d -> d.getType().equals("Available") && d.getStatus().equals("True"));
```

Comment:
@lucamolteni should I change the format of this conditio? Is here a difference between v1 and v2?

Reply:
Yes there's no deployment for processor. Each ManagedProcessor resource will provision a CamelIntegration, and that one as a deployment